### ISSUE TITLE
Animated Layout not working with react navigation (2.3.0-alpha.1)

### ISSUE THREAD

**Issue by hngocl** [software-mansion/react-native-reanimated#2124]
i try run app with react navigation then get this error. but still working without react navigation

![image](https://user-images.githubusercontent.com/43195241/121291534-a2089d80-c912-11eb-9a77-65be44471cf8.png)

### Expected behavior
Run app without error

## Package versions

- React Native: 0.64.1
- React Native Reanimated: 2.3.0-alpha.1
- React Navigation Native: 5.9.4
- React Navigation Stack: 5.14.5
- Xcode: 12.4

## Affected platforms

- [ ] Android
- [x] iOS
- [ ] Web

**Comment by SimpleCreations** [comment]
Getting this exact same error too when using `<AnimatedLayout>`. Excited to try the new API and hoping for the fix

**Comment by Szymon20000** [comment]
@hngocl @SimpleCreations Can you provide a minimal code that helps me reproduce the error? Does the issue occur only with native stacks?

**Comment by SimpleCreations** [comment]
> @hngocl @SimpleCreations Can you provide a minimal code that helps me reproduce the error? Does the issue occur only with native stacks?

https://github.com/SimpleCreations/react-native-reanimated-2124-repro

https://user-images.githubusercontent.com/12449725/122081888-2d4ccc00-ce08-11eb-9e7a-5fcd9f389568.mp4

The app still crashes when replacing `createNativeStackNavigator` with `createStackNavigator` from `@react-navigation/stack`.

**Comment by raulmax319** [comment]
Hello, I'm having the exact same problem with this setup: https://github.com/raulmax319/react-native-typescript-redux-boilerplate/tree/repro/2124_animatedLayout

**Comment by nirmaldalmia** [comment]
Do we have an update on this issue? Our release is blocked because of this issue.

**Comment by piaskowyk** [comment]
We are working on a solution, but this issue demands a lot of effort, so please be patient 🙏

**Comment by nirmaldalmia** [comment]
@piaskowyk I can understand. The issue is right into the native part of Reanimated. In the meantime, if there's a workaround or a patch please let me know.

**Comment by hannojg** [comment]
@nirmaldalmia Apply to `ios/LayoutReanimation/REASnapshooter.m`

```patch
diff --git a/node_modules/react-native-reanimated/ios/LayoutReanimation/REASnapshooter.m b/node_modules/react-native-reanimated/ios/LayoutReanimation/REASnapshooter.m
index a4666af..f6a0428 100644
--- a/node_modules/react-native-reanimated/ios/LayoutReanimation/REASnapshooter.m
+++ b/node_modules/react-native-reanimated/ios/LayoutReanimation/REASnapshooter.m
@@ -28,11 +28,13 @@ int Id = 1e9;
 
   if ([view isKindOfClass:[REAAnimationRootView class]]) {
     NSMutableArray * pathToWindow = [NSMutableArray new];
-    UIView *current = view;
+    UIResponder *current = view;
     do {
-      [pathToWindow addObject:current];
-      current = current.superview;
-    } while (current != windowView);
+      if (![current isKindOfClass:[UIViewController class]]) {
+        [pathToWindow addObject:current];
+      }
+      current = current.nextResponder;
+    } while (current != windowView && current != nil);
     values[@"pathToWindow"] = pathToWindow;
   }
```

copied and slightly modified from the linked PR

**Comment by raulmax319** [comment]
> @nirmaldalmia Apply to `ios/LayoutReanimation/REASnapshooter.m`
> 
> ```diff
> diff --git a/node_modules/react-native-reanimated/ios/LayoutReanimation/REASnapshooter.m b/node_modules/react-native-reanimated/ios/LayoutReanimation/REASnapshooter.m
> index a4666af..f6a0428 100644
> --- a/node_modules/react-native-reanimated/ios/LayoutReanimation/REASnapshooter.m
> +++ b/node_modules/react-native-reanimated/ios/LayoutReanimation/REASnapshooter.m
> @@ -28,11 +28,13 @@ int Id = 1e9;
>  
>    if ([view isKindOfClass:[REAAnimationRootView class]]) {
>      NSMutableArray * pathToWindow = [NSMutableArray new];
> -    UIView *current = view;
> +    UIResponder *current = view;
>      do {
> -      [pathToWindow addObject:current];
> -      current = current.superview;
> -    } while (current != windowView);
> +      if (![current isKindOfClass:[UIViewController class]]) {
> +        [pathToWindow addObject:current];
> +      }
> +      current = current.nextResponder;
> +    } while (current != windowView && current != nil);
>      values[@"pathToWindow"] = pathToWindow;
>    }
> ```
> 
> copied and slightly modified from the linked PR. I guess it would be good to get some feedback whether this is working for everyone 🤔

So far this fix has worked for my project 😃
- "react-native": "0.64.2",
- "@react-navigation/native": "^5.9.4",
- "@react-navigation/stack": "^5.14.5",

**Comment by jmysliv** [comment]
Fixed by #2320
